src/service/rate_limit.rs

```diff
@@ -0,0 +1,105 @@
+use crate::envoy::{RateLimitDescriptor, RateLimitRequest};
+use crate::service::Service;
+use protobuf::{Message, RepeatedField};
+use proxy_wasm::hostcalls::dispatch_grpc_call;
+use proxy_wasm::types::Status;
+use std::time::Duration;
+
+const RATELIMIT_SERVICE_NAME: &str = "envoy.service.ratelimit.v3.RateLimitService";
+const RATELIMIT_METHOD_NAME: &str = "ShouldRateLimit";
+pub struct RateLimitService<'a> {
```

Comment:
I think, if `metadata` is the "headers", and in our case the tracing ones, you can just claim ownership of these `str` and get rid of `<'a>`